src/com/google/javascript/jscomp/Es6RewriteScriptsToModules.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2017 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.javascript.jscomp;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.javascript.jscomp.NodeTraversal.AbstractPreOrderCallback;
+import com.google.javascript.rhino.Node;
+import com.google.javascript.rhino.Token;
+
+/** Rewrites a script which was imported as a module into an ES6 module. */
+public final class Es6RewriteScriptsToModules extends AbstractPreOrderCallback
+    implements HotSwapCompilerPass {
+  private final AbstractCompiler compiler;
+
+  /**
+   * Creates a new Es6RewriteModules instance which can be used to rewrite ES6 modules to a
+   * concatenable form.
+   */
+  public Es6RewriteScriptsToModules(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  /** Return whether or not the given script node represents an ES6 module file. */
+  public static boolean isEs6ModuleRoot(Node scriptNode) {
```

Comment:
Looks like now we have 3 definitions of this method.  I'd probably get rid of all but one.